src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs

```diff
@@ -205,5 +217,182 @@ public static void WriteConnectionCloseError(this DiagnosticListener @this, Guid
                     });
             }
         }
+
+        public static Guid WriteTransactionCommitBefore(this DiagnosticListener @this, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
```

Comment:
The name `WriteTransactionCommitBefore`. 
WriteBeforeTransactionCommit would be more readable.